src/LitePatch/LitePatch.Services/Services/SettingsService.cs

```diff
@@ -2,59 +2,24 @@
 using System.Text.Json.Serialization;
 using LitePatch.Services.Interfaces;
 using LitePatch.Services.Models;
+using LitePatch.Services.Repo;
 
 namespace LitePatch.Services.Services;
 
 public class SettingsService : ISettingsService
 {
+    public SettingsRepository SettingsRepository { get; set; }
     public ApplicationSetting Settings { get; set; }
 
     public SettingsService()
     {
-        Settings = ReadSettings();
+        SettingsRepository = new SettingsRepository();
```

Comment:
> Please register this as a singleton in the DI Container in MainWindow.xaml.cs

Has been amended

> fetch it via dependency injection in the constructor instead of creating a new instance directly.

I used a regular constructor. I tried doing it via the new "Constructor in Class Name". But it wouldn't let me use it to read the settings into its Settings property. Is there a way to do it? 

e.g.

``` c#
public class SettingsService(ISettingsRepository settingsRepository) : ISettingsService {

public ApplicationSettings Settings { get; set = settingsRepository.Read();} 

}

```

Something like that lol